string state
---
int16 state
bool success